base-containers/base/inginious_container_api/run_student.py

```diff
@@ -81,13 +93,14 @@ def run_student(cmd, container=None,
         zmq_socket.send(msgpack.dumps({"type": "run_student", "environment": container,
                                    "time_limit": time_limit, "hard_time_limit": hard_time_limit,
                                    "memory_limit": memory_limit, "share_network": share_network,
-                                   "socket_id": socket_id}, use_bin_type=True))
+                                   "socket_id": socket_id, "ssh": ssh}, use_bin_type=True))
 
         # Check if the container was correctly started
         message = msgpack.loads(zmq_socket.recv(), use_list=False, strict_map_key=False)
         assert message["type"] == "run_student_started"
+        student_container_id = message["container_id"]
 
-        # Send a dummy message to ask for retval
+        # Send a dummy message to ask for retval (this should not be removed without caution)
```

Comment:
"The socket can only work in a ping-pong way, so we need to send a dummy message"